source/grid/manifold_lib.cc

```diff
@@ -186,6 +160,88 @@ SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &sp
 }
 
 // ============================================================
+// SphericalManifold
+// ============================================================
+
+template <int dim, int spacedim>
+SphericalManifold<dim,spacedim>::SphericalManifold(const Point<spacedim> center):
+  center(center)
+{}
+
+template <int dim, int spacedim>
+Point<spacedim>
+SphericalManifold<dim,spacedim>::
+get_new_point (const Point<spacedim> &p1,
+               const Point<spacedim> &p2,
+               const double w) const
+{
+  Assert(w >=0.0 && w <= 1.0,
+         ExcMessage("w should be in the range [0.0,1.0]."));
+
+  if ( p1 == p2 ) return p1;
+
+  const Tensor<1,spacedim> v1 = p1 - center;
+  const Tensor<1,spacedim> v2 = p2 - center;
+  const double r1 = v1.norm();
+  const double r2 = v2.norm();
+
+  // Find the angle gamma described by v1 and v2:
+  const double gamma = std::acos((v1*v2)/(r1*r2));
+
+  // Find the angle sigma that correspont to archlengh equal to w
+  const double sigma = (1-w) * gamma;
+
+  // Versor with the same direction of v1
+  const Tensor<1,spacedim> t = v1/r1;
+  // Normal to v1 in the plane described by v1,v2,and the origin.
+  Tensor<1,spacedim> n = v2 - (v2*t)*t;
+  n = n/n.norm();
+
+  // Find the point Q along O,v1 such that
+  // P1,V,P2 has measure sigma.
+  const Tensor<1,spacedim> P = std::cos(sigma) * t + std::sin(sigma) * n;
+
+  // Project this point on the manifold.
+  return Point<spacedim>(center + (w*r1+(1-w)*r2)*P);
+}
+
+template <int dim, int spacedim>
+Tensor<1,spacedim>
+SphericalManifold<dim,spacedim>::
+get_tangent_vector (const Point<spacedim> &p1,
+                    const Point<spacedim> &p2) const
+{
+  Assert(p1 != p2,
+         ExcMessage("p1 and p2 should not concide."));
+
+  const double r1 = (p1 - center).norm();
+  const double r2 = (p2 - center).norm();
+  const Tensor<1,spacedim> e1 = (p1 - center)/r1;
+  const Tensor<1,spacedim> e2 = (p2 - center)/r2;
+
+  // Tangent vector to the unit sphere along the geodesic given by e1 and e2.
+  Tensor<1,spacedim> tg = (e2-(e2*e1)*e1);
+  tg = tg / tg.norm();
```

Comment:
Good point. In these case it does not work. I wrote these functions thinking to spacedim=3 dim=2. Luca ask me to generalise it to the case spacedim=3 and dim=3 but probably it is not easy as i thought.